src/wallet/rpcwallet.cpp

```diff
@@ -2002,7 +2002,7 @@ Value encryptwallet(const Array& params, bool fHelp)
     if (!EnsureWalletIsAvailable(fHelp))
         return Value::null;
 
-    auto fEnableWalletEncryption = GetBoolArg("-developerencryptwallet", false);
+    auto fEnableWalletEncryption = fExperimentalMode && GetBoolArg("-developerencryptwallet", false);
```

Comment:
Given the check [here](https://github.com/zcash/zcash/pull/2056/files#diff-c865a8939105e6350a50af02766291b7R757), this change is no longer required.